Port Applaunch

marcbonnici opened the pull request.

setrofim commented on wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/ApplaunchInterface.java:
```diff
@@ -47,3 +47,8 @@
+    /** Initialize the instrumentation for the workload */
+    public void initialize_instrumentation();
+
+}
```
This should be part of the previous commit.

setrofim commented on wa/workloads/applaunch/__init__.py:
```diff
@@ -0,0 +1,132 @@
+                  allowed_values=['launch_from_background', 'launch_from_long-idle'],
+                  description="""
+                  Choose launch_from_long-idle for measuring launch time
+                  from long-idle. These two types are described in the class
```
`class` --> `workload` here -- this is user-level documentation.

setrofim commented on wa/workloads/applaunch/__init__.py:
```diff
@@ -0,0 +1,132 @@
+    workload to support this workload, it should implement the ``ApplaunchInterface``.
+    The corresponding java file of the workload associated with the application being measured
+    is executed during the run. The application that needs to be
+    measured is passed as a parametre ``workload_name``. The parameters required for that workload
```
`parametre` --> `parameter`

setrofim commented on wa/workloads/applaunch/__init__.py:
```diff
@@ -0,0 +1,132 @@
+    This workload launches and measures the launch time of applications for supporting workloads.
+
+    Currently supported workloads are the ones that implement ``ApplaunchInterface``. For any
+    workload to support this workload, it should implement the ``ApplaunchInterface``.
```
This is more of a TO-DO note than a comment (no action required for this pull), but it would be good if we could auto-generate a list of these; maybe with decompiliation/introspection tools in the Android SDK....

setrofim commented on wa/workloads/applaunch/__init__.py:
```diff
@@ -0,0 +1,132 @@
+    The corresponding java file of the workload associated with the application being measured
+    is executed during the run. The application that needs to be
+    measured is passed as a parametre ``workload_name``. The parameters required for that workload
+    have to be passed as a dictionary which is captured by the parametre ``workload_params``.
```
`parametre` --> `parameter`

setrofim commented on wa/workloads/applaunch/__init__.py:
```diff
@@ -0,0 +1,132 @@
+    ]
+
+    def __init__(self, device, **kwargs):
+        super(Applaunch, self).__init__(device, **kwargs)
```
This is redundant is it just class the super. (Also `device` --> `target` to match WA3 nomenclature, but that's irrelevant as the mehtod should be removed).

setrofim commented on wa/workloads/applaunch/__init__.py:
```diff
@@ -0,0 +1,132 @@
+
+        self.pass_parameters()
+
+
```
Too many blank lines

setrofim commented on wa/workloads/googlephotos/__init__.py:
```diff
@@ -70,7 +70,7 @@ def init_resources(self, context):
+    def deploy_assets(self, context):
         super(Googlephotos, self).deploy_assets(context)
         # Create a subfolder for each test_image named ``wa-[1-4]``
         # Move each image into its subfolder
```
This commit only modifies Python source, yet has an APK attached. Should the APK be part of the previous commit?

marcbonnici commented:
Addressed comments above.
Additionally:
  - Added commits to alter the way assets are deployed to be able to be used more generically. 
  - Applaunch now deploys required uiauto apk files as assets.
  - Fixed applaunch not cleaning up after itself properly.
